### PR TITLE
[executor] array support

### PR DESCRIPTION
**Pull request by JonaZz747**
- Datastructures (on locals, stack, ...) + opCodes for
  - primitive Arrays
  - Object Arrays
  - Multidim Arrays

### CODE REVIEW

**Comment by liketechnik** [comment]
In it's current state this does not cover all array opcodes (in particular most opcodes dealing with primitve datatype arrays + `newarray` + `multianewarray` are still missing), but the stuff that's implemented is already finished.

TL~DR: this can be merged now, if you don't have anything to (re)consider during review. (So that the parser team can use the changed opcode definitions)

File: src/class/builtin_classes/array.rs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+pub type DoubleArray = Array<DoubleArrayKind>;
+pub type DoubleArrayInstance = ArrayInstance<DoubleArrayKind>;
+#[derive(Default)]
+pub struct DoubleArrayKind {}
+
+impl ArrayKind for DoubleArrayKind {
+    type Value = f64;
+
+    fn class_name(&self) -> &str {
+        "[C"
```
Comment (Aetlam) [inline src/class/builtin_classes/array.rs:99]: This seems like a coppy paste error should be [D

File: src/class/builtin_classes/array.rs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+pub type FloatArray = Array<FloatArrayKind>;
+pub type FloatArrayInstance = ArrayInstance<FloatArrayKind>;
+#[derive(Default)]
+pub struct FloatArrayKind {}
+
+impl ArrayKind for FloatArrayKind {
+    type Value = f64;
+
+    fn class_name(&self) -> &str {
+        "[C"
```
Comment (Aetlam) [inline src/class/builtin_classes/array.rs:116]: same here

**Review by Aetlam — CHANGES_REQUESTED** [review]

**Comment by liketechnik** [comment]
Failures kommen durch den merge, ich kümmere mich gleich um den rebase

**Pull request merged**